src/main/java/graphql/normalized/ExecutableNormalizedOperationFactory.java

```diff
@@ -333,6 +386,13 @@ private ExecutableNormalizedOperation createNormalizedQueryImpl(GraphQLSchema gr
             normalizedFieldToMergedField.put(enf, mergedFld);
         };
 
+        ImmutableListMultimap.Builder<ExecutableNormalizedField, DeferExecution> normalizedFieldToDeferExecution = ImmutableListMultimap.builder();
+        normalizedFieldToDeferExecution.putAll(collectFromOperationResult.normalizedFieldToDeferExecution);
+
+        Consumer<CollectNFResult> captureCollectNFResult = (collectNFResult -> {
```

Comment:
I feel like working with an instance of some object created on a per-execution basis would make accessing these values much easier.  We could create instance fields instead of having to pass them down the stack.